### PR TITLE
add unit test for ZuulEndPointRunner

### PR DESCRIPTION
**Pull request by sullis**

### CODE REVIEW

File: zuul-core/src/test/java/com/netflix/zuul/netty/filter/ZuulEndPointRunnerTest.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+    @Test
+    public void nonErrorEndpoint() {
+        request.getContext().setShouldSendErrorResponse(false);
+        request.getContext().setEndpoint(BASIC_ENDPOINT);
+        assertNull(request.getContext().get(ZUUL_ENDPOINT));
+        endpointRunner.filter(request);
+        final ZuulFilter filter = request.getContext().get(ZUUL_ENDPOINT);
+        assertTrue(filter instanceof BasicEndpoint);
+        verify(filterRunner, times(1)).filter(any());
```
Comment (argha-c) [inline zuul-core/src/test/java/com/netflix/zuul/netty/filter/ZuulEndPointRunnerTest.java:98]: This should at least assert the request message setup above, was passed in.

File: zuul-core/src/test/java/com/netflix/zuul/netty/filter/ZuulEndPointRunnerTest.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        when(chc.executor()).thenReturn(ImmediateEventExecutor.INSTANCE);
+        context.put(NETTY_SERVER_CHANNEL_HANDLER_CONTEXT, chc);
+        request = new HttpRequestMessageImpl(context, "http", "GET", "/foo/bar", new HttpQueryParams(), headers, "127.0.0.1", "http", 8080, "server123");
+        request.storeInboundRequest();
+    }
+
+    @Test
+    public void nonErrorEndpoint() {
+        request.getContext().setShouldSendErrorResponse(false);
+        request.getContext().setEndpoint(BASIC_ENDPOINT);
```
Comment (argha-c) [inline zuul-core/src/test/java/com/netflix/zuul/netty/filter/ZuulEndPointRunnerTest.java:93]: To test for the real flow, or at least part of it, this should really set the `KEY_ENDPOINT` state on the context.
Setting the endpoint directly defeats the purpose of testing the part where the endpoint is looked up by the runner

**Review by argha-c — CHANGES_REQUESTED** [review]

File: zuul-core/src/test/java/com/netflix/zuul/netty/filter/ZuulEndPointRunnerTest.java (hunk 2 above)
Comment (sullis) [reply zuul-core/src/test/java/com/netflix/zuul/netty/filter/ZuulEndPointRunnerTest.java]: The test now invokes the set method on the context:

```
request.getContext().set("_endpoint", BASIC_ENDPOINT);
```

I could not use the KEY_ENDPOINT variable.  (KEY_ENDPOINT is a private static variable)

File: zuul-core/src/test/java/com/netflix/zuul/netty/filter/ZuulEndPointRunnerTest.java (hunk 1 above)
Comment (sullis) [reply zuul-core/src/test/java/com/netflix/zuul/netty/filter/ZuulEndPointRunnerTest.java]: I added a Mockito ArgumentCaptor. I am now making assertions against the captured value.

File: zuul-core/src/test/java/com/netflix/zuul/netty/filter/ZuulEndPointRunnerTest.java (hunk 2 above)
Comment (argha-c) [reply zuul-core/src/test/java/com/netflix/zuul/netty/filter/ZuulEndPointRunnerTest.java]: `setEndpoint` on context should work though?

Comment (sullis) [reply zuul-core/src/test/java/com/netflix/zuul/netty/filter/ZuulEndPointRunnerTest.java:92]: Ack.  This works for me:
```
        request.getContext().setEndpoint(BASIC_ENDPOINT);
```

**Review by argha-c — APPROVED** [review]

**Review by argha-c — APPROVED** [review]

**Comment by sullis** [comment]
LGTM. Ready to merge.

**Pull request merged**